flareon/src/db.rs

```diff
@@ -318,18 +320,72 @@ pub trait FromDbValue {
         Self: Sized;
 }
 
+pub type DbValue = sea_query::Value;
+
 /// A trait for converting a Rust value to a database value.
 pub trait ToDbValue: Send + Sync {
     /// Converts the Rust value to a `sea_query` value.
     ///
     /// This method is used to convert the Rust value to a value that can be
     /// used in a query.
-    fn to_sea_query_value(&self) -> sea_query::Value;
+    fn to_db_value(&self) -> DbValue;
+}
+
+/// A generalization of [`ToDbValue`] that can also return a marker that means a
+/// value should be automatically generated by the database.
+pub trait ToDbFieldValue {
+    fn to_db_field_value(&self) -> DbFieldValue;
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum DbFieldValue {
+    /// The value should be automatically generated by the database and not
+    /// included in the query.
+    Auto,
+    /// A value that should be included in the query.
+    Value(DbValue),
+}
+
+impl DbFieldValue {
+    #[must_use]
+    pub fn is_auto(&self) -> bool {
+        matches!(self, Self::Auto)
+    }
+
+    #[must_use]
+    pub fn is_value(&self) -> bool {
+        matches!(self, Self::Value(_))
+    }
+
+    #[must_use]
+    pub fn unwrap_value(self) -> sea_query::Value {
+        self.expect_value("called DbValue::unwrap_value() on a wrong DbValue variant")
+    }
+
+    #[must_use]
+    pub fn expect_value(self, message: &str) -> sea_query::Value {
```

Comment:
For consistency sake
```suggestion
    pub fn unwrap_value(self) -> DbValue {
        self.expect_value("called DbValue::unwrap_value() on a wrong DbValue variant")
    }

    #[must_use]
    pub fn expect_value(self, message: &str) -> DbValue {
```

Reply:
Fair!